WalletWasabi/WabiSabi/Client/CoinJoinCoinSelector.cs

```diff
@@ -38,26 +39,26 @@ public class CoinJoinCoinSelector
 	public int AnonScoreTarget { get; }
 	public int SemiPrivateThreshold { get; }
 	private WasabiRandom Rnd { get; }
-	
+
 	public static CoinJoinCoinSelector FromWallet(IWallet wallet) =>
-		new (
+		new(
 			wallet.ConsolidationMode,
 			wallet.AnonScoreTarget,
 			wallet.RedCoinIsolation ? Constants.SemiPrivateThreshold : 0,
 			SecureRandom.Instance);
-	
+
 	/// <param name="liquidityClue">Weakly prefer not to select inputs over this.</param>
 	public ImmutableList<TCoin> SelectCoinsForRound<TCoin>(IEnumerable<TCoin> coins, UtxoSelectionParameters parameters, Money liquidityClue)
 		where TCoin : class, ISmartCoin, IEquatable<TCoin>
 	{
 		liquidityClue = liquidityClue > Money.Zero
 			? liquidityClue
 			: Constants.MaximumNumberOfBitcoinsMoney;
-			
+
 		var filteredCoins = coins
 			.Where(x => parameters.AllowedInputAmounts.Contains(x.Amount))
 			.Where(x => parameters.AllowedInputScriptTypes.Contains(x.ScriptType))
-			.Where(x => x.EffectiveValue(parameters.MiningFeeRate) > Money.Zero)
+			.Where(x => x.EffectiveValue(parameters.MiningFeeRate).Satoshi > x.Amount.Satoshi * (1 - MaxEffectiveValueLossTolerance)
```

Comment:
Suggestion. Move this to somewhere else to make it clearer.
```suggestion
			.Where(x => IsEconomicallyuFeasible(x))
```

```c#
static IsEconomicallyuFeasible(Coin coin, FeeRate miningFeeRate) =>
    coin.EffectiveValue(miningFeeRate) > coin.Amount * (1.0 - MaxEffectiveValueLossTolerance);
```